lib/puppet/parser/scope.rb

```diff
@@ -738,10 +815,12 @@ def to_s
 
   alias_method :inspect, :to_s
 
-  # remove ephemeral scope up to level
-  # TODO: Who uses :all ? Remove ??
+  # Pop ephemeral scopes up to level and return them
   #
+  # @deprecated use #pop_epehemeral
+  # @api private
   def unset_ephemeral_var(level=:all)
+    Puppet.deprecation_warning('Method Parser::Scope#unset_ephemeral_var() is deprecated')
```

Comment:
I decided to deprecate that method as a whole since you also wanted me to add an alias `pop_ephemerals`for it. All calls, except from the ones using :all (only from tests) have been replaced to use `pop_ephemerals` so I think it's safe to say that this method (@api private as it is) is indeed deprecated.